src/mesh/FloodingRouter.cpp

```diff
@@ -90,14 +93,67 @@ void FloodingRouter::perhapsRebroadcast(const meshtastic_MeshPacket *p)
             if (isRebroadcaster()) {
                 meshtastic_MeshPacket *tosend = packetPool.allocCopy(*p); // keep a copy because we will be sending it
 
-                tosend->hop_limit--; // bump down the hop count
-#if USERPREFS_EVENT_MODE
-                if (tosend->hop_limit > 2) {
-                    // if we are "correcting" the hop_limit, "correct" the hop_start by the same amount to preserve hops away.
-                    tosend->hop_start -= (tosend->hop_limit - 2);
-                    tosend->hop_limit = 2;
+                // First hop MUST always decrement to prevent retry issues
+                bool isFirstHop = (p->hop_start != 0 && p->hop_start == p->hop_limit);
+                
+                // Check if both local device and previous relay are routers (including CLIENT_BASE)
+                bool localIsRouter = IS_ONE_OF(config.device.role, meshtastic_Config_DeviceConfig_Role_ROUTER,
+                                                meshtastic_Config_DeviceConfig_Role_ROUTER_LATE,
+                                                meshtastic_Config_DeviceConfig_Role_CLIENT_BASE);
+                
+                bool prevRelayIsRouter = false;
+                
+                // Only check for router-to-router on subsequent hops (not first hop)
+                if (!isFirstHop) {
```

Comment:
Please also don't loop through the NodeDB when we are not a router (`!localIsRouter`).

Reply:
if (!localIsRouter) {
        return true;
    }
    
   Added to Router::shouldDecrementHopLimit